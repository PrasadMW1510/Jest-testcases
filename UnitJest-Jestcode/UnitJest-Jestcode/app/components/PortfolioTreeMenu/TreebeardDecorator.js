/**
 *
 * PortfolioTreeMenu
 *
 */

import React from 'react';
import PropTypes from 'prop-types';

import folderImage from 'images/yellow_folder.png';
import OpenFolderImage from 'images/open_yellow_folder.png';
import fileImage from 'images/file.png';

const TreebeardDecorator = ({ style, node }) => {
  if (node.name === 'Loading') {
    return node.name;
  }
  const iconType = node.children ? 'folder' : 'file-text';
  const iconClass = `fa fa-${iconType}`;
  let nodeIcon = null;
  switch (iconType) {
    case 'folder':
      nodeIcon = node.toggled ? OpenFolderImage : folderImage;
      break;
    default:
      nodeIcon = fileImage;
      break;
  }
  return (
    <div style={style.base} className="portfolio-treenode">
      <div style={style.title} className="portfolio-treenode">
        <i className={iconClass}>
          <img style={{ width: '18px' }} alt="" src={nodeIcon} />
        </i>
        {node.name}
      </div>
    </div>
  );
};

TreebeardDecorator.propTypes = {
  style: PropTypes.object,
  node: PropTypes.object,
};

export default TreebeardDecorator;
