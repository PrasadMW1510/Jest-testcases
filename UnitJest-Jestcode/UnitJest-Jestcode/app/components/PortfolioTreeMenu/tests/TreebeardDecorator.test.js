import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import TreebeardDecorator from '../TreebeardDecorator';

describe('<TreebeardDecorator  />', () => {
  let wrapper = null;
  let wrapper1 = null;
  let wrapper2 = null;
  let wrapper3 = null;

  const props = {
    style: {},
    node: {
      name: 'Loading',
      toggled: true,
    },
  };
  const props1 = {
    style: {},
    node: {
      name: 'Loadin',
      children: [],
      toggled: true,
    },
  };
  const props3 = {
    style: {},
    node: {
      name: 'Loadin',
      children: [],
    },
  };
  const props2 = {
    style: {},
    node: {
      name: 'Loadin',
      toggled: true,
    },
  };
  wrapper = shallow(<TreebeardDecorator {...props} />);
  wrapper1 = shallow(<TreebeardDecorator {...props1} />);
  wrapper2 = shallow(<TreebeardDecorator {...props2} />);
  wrapper3 = shallow(<TreebeardDecorator {...props3} />);

  it('Expect to render correctly', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
    expect(shallowToJson(wrapper1)).toMatchSnapshot();
    expect(shallowToJson(wrapper2)).toMatchSnapshot();
    expect(shallowToJson(wrapper3)).toMatchSnapshot();
  });
});
