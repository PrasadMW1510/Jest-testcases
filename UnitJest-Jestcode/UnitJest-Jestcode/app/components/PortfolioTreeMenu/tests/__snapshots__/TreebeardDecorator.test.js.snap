// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TreebeardDecorator  /> Expect to render correctly 1`] = `"Loading"`;

exports[`<TreebeardDecorator  /> Expect to render correctly 2`] = `
<div
  className="portfolio-treenode"
>
  <div
    className="portfolio-treenode"
  >
    <i
      className="fa fa-folder"
    >
      <img
        alt=""
        src="IMAGE_MOCK"
        style={
          Object {
            "width": "18px",
          }
        }
      />
    </i>
    Loadin
  </div>
</div>
`;

exports[`<TreebeardDecorator  /> Expect to render correctly 3`] = `
<div
  className="portfolio-treenode"
>
  <div
    className="portfolio-treenode"
  >
    <i
      className="fa fa-file-text"
    >
      <img
        alt=""
        src="IMAGE_MOCK"
        style={
          Object {
            "width": "18px",
          }
        }
      />
    </i>
    Loadin
  </div>
</div>
`;

exports[`<TreebeardDecorator  /> Expect to render correctly 4`] = `
<div
  className="portfolio-treenode"
>
  <div
    className="portfolio-treenode"
  >
    <i
      className="fa fa-folder"
    >
      <img
        alt=""
        src="IMAGE_MOCK"
        style={
          Object {
            "width": "18px",
          }
        }
      />
    </i>
    Loadin
  </div>
</div>
`;
